src/fetchers/book.ts

```diff
@@ -0,0 +1,39 @@
+import { useQuery } from '@tanstack/react-query';
+import { Position } from '@penumbra-zone/protobuf/penumbra/core/component/dex/v1/dex_pb';
+import { JsonValue } from '@bufbuild/protobuf';
+
+interface BookResponse {
+  asks: Position[];
+  bids: Position[];
+}
+
+interface BookResponseJson {
+  asks: JsonValue[];
+  bids: JsonValue[];
+}
+
+export const useBook = (
+  symbol1: string | undefined,
+  symbol2: string | undefined,
+  hops: number | undefined,
+  limit: number | undefined,
+) => {
+  return useQuery({
+    queryKey: ['book', symbol1, symbol2, hops, limit],
+    queryFn: async (): Promise<BookResponse> => {
+      if (!symbol1 || !symbol2 || !limit) {
```

Comment:
question: limit is a required field too? If so, think we should change the type signature of the hook.